internal/controller/controllers/agent_controller.go

```diff
@@ -131,10 +131,14 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 	if err = r.Get(ctx, kubeKey, clusterDeployment); err != nil {
 		if k8serrors.IsNotFound(err) {
 			// Delete the agent, using a finalizer with pre-delete to deregister the host.
+			log.Infof("Cluster Deployment name: %s namespace: %s not found, deleting Agent",
```

Comment:
hmm..
Not a comment about this log but, now that I look at it, maybe this should actually try to fetch ACI and not CD? as we do clear up agents per ACI Finalizer.

Reply:
@filanov what do you think? ☝️

Reply:
maybe both, there maybe a case that only ACI is deleted right?